lib/utils.js

```diff
@@ -97,6 +97,8 @@ var Utils = module.exports = {
     return lodash.cloneDeep(obj, function (elem) {
       // Unfortunately, lodash.cloneDeep doesn't preserve Buffer.isBuffer, which we have to rely on for binary data
       if (Buffer.isBuffer(elem)) { return elem; }
+      // Preserve data-types like `fn` across clones
+      if (elem instanceof Utils.fn) { return new Utils.fn(elem.fn, elem.args); }
```

Comment:
Since this is a general-purpose deepClone method I would probably expect all properties to be cloned &  that any changes made to one object wouldn't be reflected on its clone.